### ISSUE TITLE
How about add AssemblyScript output target?

### ISSUE THREAD

**Issue by MaxGraey** [esmbly/esmbly#4]
It will be great if user will have ability transform existing  JS, pure TS, flow and jsdoc to AssemblyScript subset as well as direct to wasm/wat target. wdyt?

**Comment by ollelauribostrom** [comment]
@MaxGraey  Sounds like a good idea 🙂 

Each transformer is responsible for two things (these steps are currently optional),
1. Transform the AST from the input format to the output format `.transform()`
2. Create the desired output files `.createFiles()`

The `wasm-transformer` (currently WIP) does not perform any transformations on the AST. It only implements the createFiles method which basically just calls the AssemblyScript compiler with different flags depending on the desired output (.wasm, .wat, .asm.js).

I'm guessing that we could add the `transform()` method as well, and transform the AST to the AssemblyScript subset before creating the output files. Am I correct to assume that [this page](https://github.com/AssemblyScript/assemblyscript/wiki/Limitations) explains the differences between TS and AS that needs to be transformed/checked?

**Comment by MaxGraey** [comment]
> Am I correct to assume that this page explains the differences between TS and AS that needs to be transformed/checked

Yes. Anyway you should transform to AS first if you use AS as middleware between js and wasm.
Difference between TS and AS:
1. any function/method should return type explicitly
2. `any`, `undefined`, `delete` is not supported and never be supported.
3. AS still in progress and has lack handling of `interface` and `complex union types`. plain types can't be nullable yet, but reference types like `arr: number[] | number` are already support. Also see about runtime status: https://github.com/AssemblyScript/assemblyscript/wiki/Status-and-Roadmap#standard-library-compatibility
